### ISSUE TITLE
Combatible with cake php v4.1

### ISSUE THREAD

**Issue by hkosmidi** [Holt59/cakephp3-bootstrap-helpers#194]
This is a (multiple allowed):

* [ ] bug
* [x] enhancement
* [ ] question

* CakePHP Version: 4.6.1

### What you did
I used your plugin with cake php v4.1 but got several errors

### What happened
Some functions didn't have proper return type in deceleration and yield errors

### What you expected to happen
I have updated the function calls in 2 commits. Could you please check if they are ok. Is there way to merge in your codebase so to use your plugin through composer?
https://github.com/hkosmidi/cakephp3-bootstrap-helpers/commit/9c48d6591bc9f47a2bba4313a8a272d718f593a9
https://github.com/hkosmidi/cakephp3-bootstrap-helpers/commit/3ba820d2e923f28472e51b42be71fedf9cc5c26c

**Comment by Erwane** [comment]
did you try this branch ? ` https://github.com/Holt59/cakephp3-bootstrap-helpers/tree/cake4-bs4-1.0.0`

**Comment by hkosmidi** [comment]
No I haven't. I use bootstrap 3 for now

**Comment by Erwane** [comment]
can you make a PR ? with tests ?
or i can do it if you prefer

**Comment by hkosmidi** [comment]
Sorry I have not made any test, I just added the appropriate return types in functions

**Comment by Holt59** [comment]
I've created https://github.com/Holt59/cakephp3-bootstrap-helpers/tree/cake4-bs3-1.0.0 with your two commits. This is 100% untested, I've only cherry-picked your two commits.

You can require it using `"holt59/cakephp3-bootstrap-helpers": "dev-cake4-bs3-1.0.0"`.

**Comment by Erwane** [comment]
I'm trying different notation for this project but i don't know how to do to maintain 4 version with comprehensive tags.

one way is 
4 branches (cakeX-bsY) with tag attached like this : 
4.3.1 is cake3-bs3 version 1.
The risk is a "^4.3" dependency in `composer.json` apps and the require should be `~4.3.0` to load cake4-bs3 and all fixes

One alternative is tag `43.0.0` for cake4-bs3, `33.0.0`  and write documentation to require `^43.0`

**Comment by Holt59** [comment]
I've been thinking about it, and there are multiple issues... I think the best option would be to fork this to a `cakephp4-bootstrap-helpers`.

We could keep all development here, in custom branch names such as the ones we are using, and only push tags to the `cakephp4-` repository so there is no issue with composer.

**Comment by Erwane** [comment]
Ok, looks better but what about bootstrap versions ?
version 3.xx for bs3 and 4.xx for bs4 ?

**Comment by Holt59** [comment]
Yes, 3.x.x for bs3, 4.x.x for bs4, etc. It's what's already done.

**Comment by Erwane** [comment]
So here, you keep branches : `cake3-bs3`, `cake4-bs3`n `cake4-bs4`, etc etc 
maintain tags for cakephp3 with bootstrap 3 and 4 in `3.x.y` and `4.x.y`
And push the dev in `cake4-bsX` branches to new repository `cakephp4-bootstrap-helpers` with tags `3.x` and `4.x` for bootstrap versions ?

No PR in the `cake4-bootstrap-helpers` repository ? This part sound weird.
Logic is to keep cake4 dev in this new repository, and push fixes from cake3 to cake4 no ?

**Comment by Holt59** [comment]
Well, I'd say its a preference choice, I don't really mind:
- Either we keep all development here and only push to the Cake 4 repository.
- Or we separate development, Cake 3 here and Cake 4 on the other repository.

We could completely separate development and only cherry-pick when needed.

I've created the Cake 4 repository and registered it on packagist: https://github.com/Holt59/cakephp4-bootstrap-helpers (the v3.0.0 and v4.0.0 corresponds to the cake4-bs3 and cake4-bs4 branches here).

I've also added you as collaborator to both repository so you should receive an invite.

**Comment by Holt59** [comment]
Because that's what it was called here :) But it should be v3.0.x and we should have a v3.x branch:

- v3.x &mdash; Latest development branch for Bootstrap 3.
- v3.0.x &mdash; Bug fixes for release v3.0.x (created after v3.0.0 is released).
- Similar branches for Bootstrap 4.

**Comment by hkosmidi** [comment]
I also added this commit https://github.com/hkosmidi/cakephp3-bootstrap-helpers/commit/8b0d5db39c7b6c6c7a3065d13a8aaf203bc02b5a  in order for DateTimeWidget to play along. Hope this helps

**Comment by Erwane** [comment]
The cake4 support will be moved to this repository. https://github.com/Holt59/cakephp4-bootstrap-helpers
And this PR should fix Helpers initialization https://github.com/Holt59/cakephp4-bootstrap-helpers/pull/3

Waiting for @Holt59
